app/charts/pie/pie-state.tsx

```diff
@@ -158,8 +144,19 @@ const usePieState = (
       colors.domain(segments);
       colors.range(getPalette(fields.segment?.palette));
     }
+    // Do not let the scale be implicitly extended by children calling it
+    // on unknown values
+    colors.unknown(() => undefined);
```

Comment:
Do you know in which cases can this happen?

Reply:
Not really, I just noticed that we had extra entries in the legend 🤷‍♂️